admin/RELEASING.md

```diff
@@ -0,0 +1,18 @@
+# How-to release `rustls-platform-verifier`
```

Comment:
WDYT about adding some of the text from the Rustls [RELEASING.md](https://github.com/rustls/rustls/blob/main/RELEASING.md#maintenance-point-releases) that talks about proposing release notes?

Do you think using GitHub releases would also be helpful? I think it's a good way to help the many Dependabot users.

Reply:
I'm in favor of using GitHub releases. I omitted a bit from these exact steps but in other repositories I maintain I use GitHub releases to create both a release and tag at the same time. After creating the tag via release, I check it out and run `cargo publish`. I'll try adding more specifics here.